src/Microsoft.Extensions.Configuration.Abstractions/IConfiguration.cs

```diff
@@ -6,12 +6,29 @@
 
 namespace Microsoft.Extensions.Configuration
 {
+    /// <summary>
+    /// Represents a set of key/value application configuration properties.
+    /// </summary>
     public interface IConfiguration
     {
+        /// <summary>
+        /// Gets or sets a configuration value.
+        /// </summary>
+        /// <param name="key">The configuration key.</param>
+        /// <returns>The configuration value.</returns>
         string this[string key] { get; set; }
 
+        /// <summary>
+        /// Gets a configuration sub-section with the specified key.
```

Comment:
Sounds like a good thing to put in `<remarks/>` if we are using that. 

The good thing is that your empty section can be populated on reload if any new data matches the path :smile: